website/server/models/user/methods.js

```diff
@@ -91,3 +98,17 @@ schema.methods.addComputedStatsToJSONObj = function addComputedStatsToUserJSONOb
 
   return statsObject;
 };
+
+schema.methods.cancelSubscription = async function cancelSubscription () {
+  let plan = this.purchased.plan;
+
+  if (plan.paymentMethod === amazonPayments.constants.PAYMENT_METHOD_AMAZON) {
+    return await amazonPayments.cancelSubscription({user: this});
+  } else if (plan.paymentMethod === stripePayments.constants.PAYMENT_METHOD) {
+    return await stripePayments.cancelSubscription({user: this});
+  } else if (plan.paymentMethod === paypalPayments.constants.PAYMENT_METHOD) {
+    return await paypalPayments.subscribeCancel({user: this});
+  }
+
+  return await payments.cancelSubscription({user: this});
```

Comment:
so payments. cancelSubscription should not be called directly anymore? I'm a little confused by this: we have user.cancelSubscription, payments.cancelSubscrition({user}) and all the payments specific ones and all do a different thing. For example what's the difference between `paypalPayments.subscribeCancel({user: this});` and `payments.cancelSubscription()` for a user that uses paypal?